tiled/_tests/test_awkward.py

```diff
@@ -3,116 +3,172 @@
+    # named like 'node0-offsets' and 'node2-data'.
+    array = awkward.Array(
+        [
+            [{"x": 1.1, "y": [1]}, {"x": 2.2, "y": [1, 2]}],
+            [],
+            [{"x": 3.3, "y": [1, 2, 3]}],
+        ]
+    )
+    aac = client.write_awkward(array, key="test")
+
+    file = io.BytesIO()
+    aac.export(file, format="application/json")
+    actual = bytes(file.getbuffer()).decode()
+    assert actual == awkward.to_json(array)
+
+
+def test_export_arrow(tmpdir, client):
+    # Write data into catalog. It will be stored as directory of buffers
+    # named like 'node0-offsets' and 'node2-data'.
+    array = awkward.Array(
+        [
+            [{"x": 1.1, "y": [1]}, {"x": 2.2, "y": [1, 2]}],
+            [],
+            [{"x": 3.3, "y": [1, 2, 3]}],
+        ]
+    )
+    aac = client.write_awkward(array, key="test")
+
+    filepath = tmpdir / "actual.arrow"
+    aac.export(str(filepath), format=APACHE_ARROW_FILE_MIME_TYPE)
+    actual = pyarrow.feather.read_table(filepath)
+    expected = awkward.to_arrow_table(array)
+    assert actual == expected
+
+
+def test_export_parquet(tmpdir, client):
+    # Write data into catalog. It will be stored as directory of buffers
+    # named like 'node0-offsets' and 'node2-data'.
+    array = awkward.Array(
+        [
+            [{"x": 1.1, "y": [1]}, {"x": 2.2, "y": [1, 2]}],
+            [],
+            [{"x": 3.3, "y": [1, 2, 3]}],
+        ]
+    )
+    aac = client.write_awkward(array, key="test")
+
+    filepath = tmpdir / "actual.parquet"
+    aac.export(str(filepath), format="application/x-parquet")
+    actual = pyarrow.parquet.read_table(filepath)
+    expected = awkward.to_arrow_table(array)
+    assert actual == expected
+
+
+@pytest.fixture
+def spy():
+    def mock_request(method, *args, **kwargs):
+        return method(*args, **kwargs)
+
+    spy = Mock(side_effect=mock_request)
+    yield spy
+
+
+@pytest.fixture
+def spy_client(client, spy):
+    original_http_client = client.context.http_client
+
+    spy_client = Mock(create=True)
+    # spy_client.__getattr__ = lambda attr_name: getattr(client, attr_name)
+
+    spy_client.get = Mock(side_effect=partial(spy, original_http_client.get))
+    spy_client.put = Mock(side_effect=partial(spy, original_http_client.put))
+    spy_client.post = Mock(side_effect=partial(spy, original_http_client.post))
+
+    client.context.http_client = spy_client
+    yield client
+
+    client.context.http_client = original_http_client
+
+
+@pytest.mark.parametrize(
+    "browser, url_length_limit", (
+        # ("Chrome", 2_083),
+        # ("Firefox", 65_536),
+        ("Safari", 80_000),
+        # ("Internet Explorer", 2_083),
+    )
+)
+def test_long_url(spy_client, spy, url_length_limit, browser, num_keys=7_000):
+    # https://github.com/bluesky/tiled/pull/577
+    array = awkward.Array([{f"key{i:05}": i for i in range(num_keys)}])
+    aac = spy_client.write_awkward(array, key="test")
+    aac[0]
+
```

Comment:
I added a check on the form_keys length here.  It's a bit invasive on the implementation, but this is one way to show what is being tested here.

Reply:
To make this change simpler, I refactored the tests to use a client fixture.

Feel free to pop this commit.  I added it just to show my thinking.